specification/security/resource-manager/Microsoft.Security/preview/2024-07-01-preview/securityConnectors.json

```diff
@@ -0,0 +1,1707 @@
+              "x-ms-enum": {
+                "name": "scanningMode",
+                "modelAsString": true
+              }
+            },
+            "exclusionTags": {
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              },
+              "description": "Tags that indicates that a resource should not be scanned"
+            }
+          }
+        }
+      }
+    },
+    "vmScannersGcp": {
+      "type": "object",
+      "description": "A VM scanning configuration for a security offering of a GCP environment",
+      "properties": {},
+      "allOf": [
+        {
+          "$ref": "#/definitions/vmScannersBase"
+        }
+      ]
+    },
+    "vmScannersAws": {
+      "type": "object",
+      "description": "A VM scanning configuration for a security offering of a Aws environment",
+      "properties": {
+        "cloudRoleArn": {
+          "type": "string",
+          "description": "The cloud role ARN in AWS for this feature"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/vmScannersBase"
+        }
+      ]
+    },
+    "arcAutoProvisioning": {
+      "type": "object",
+      "description": "The ARC autoprovisioning configuration",
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "Is arc auto provisioning enabled"
+        },
+        "configuration": {
+          "type": "object",
+          "description": "Configuration for servers Arc auto provisioning for a given environment",
+          "properties": {
+            "proxy": {
+              "type": "string",
+              "description": "Optional HTTP proxy endpoint to use for the Arc agent"
+            },
+            "privateLinkScope": {
+              "type": "string",
+              "description": "Optional Arc private link scope resource id to link the Arc agent"
+            }
+          }
+        }
+      }
+    },
+    "arcAutoProvisioningGcp": {
+      "type": "object",
+      "description": "The ARC autoprovisioning configuration for an GCP environment",
+      "properties": {},
+      "allOf": [
+        {
+          "$ref": "#/definitions/arcAutoProvisioning"
+        }
+      ]
+    },
+    "arcAutoProvisioningAws": {
+      "type": "object",
+      "description": "The ARC autoprovisioning configuration for an AWS environment",
+      "properties": {
+        "cloudRoleArn": {
+          "type": "string",
+          "description": "The cloud role ARN in AWS for this feature"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/arcAutoProvisioning"
+        }
+      ]
+    }
+  },
+  "parameters": {
+    "securityConnectorName": {
+      "name": "securityConnectorName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The security connector name.",
+      "x-ms-parameter-location": "method"
+    },
+    "securityConnectorInBody": {
```

Comment:
why a separate parameter then ? can we not use definitions/SecurityConnector directly ?

Reply:
This version is build on top of the older version with the goal of supporting a new environment only, without the intention to make further unrelated changes